src/cc/bcc_syms.cc

```diff
@@ -92,51 +95,112 @@ bool KSyms::resolve_name(const char *_unused, const char *name,
   return true;
 }
 
-ProcSyms::ProcSyms(int pid) : pid_(pid), procstat_(pid) { load_modules(); }
+ProcMountNS::ProcMountNS(int pid) : self_fd_(-1), target_fd_(-1) {
+  if (pid < 0)
+    return;
+
+  int self_fd = -1;
```

Comment:
Can I suggest using the FileDesc class from src/cc/table_desc.h? You would have a local `FileDesc self_fd`, that at the end of the function gets `self_fd_ = std::move(self_fd)`.